src/rockstor/system/osi.py

```diff
@@ -786,48 +1234,136 @@ def set_disk_spindown(device, spindown_time, apm_value,
             if fields[0] == 'DEVLINKS':
-                # return the first value directly after DEVLINKS
-                return fields[1]
-    # if no DEVLINKS value found or an error occurred.
-    return None
+                # cycle through all device names on the DEVLINKS line
+                for index in range(1, len(fields)):
+                    # check if device name is by-id type
+                    if re.match('/dev/disk/by-id', fields[index]) is not None:
+                        is_byid = True
+                        dev_name_length = len(fields[index])
+                        # check if longer than any found previously
+                        if dev_name_length > longest_byid_name_length:
+                            longest_byid_name_length = dev_name_length
+                            # save the longest by-id type name so far.
+                            byid_name = fields[index]
+    if is_byid:
+        # Return the longest by-id name found in the DEVLINKS line
+        # or the first if multiple by-id names were of equal length.
+        return_name = byid_name
+    # Honour our path strip request in all cases if we can.
+    if remove_path:
+        # Strip the path from the beginning of our return_name.
+        # For use in Disk.name db field for example.
+        # Split return_name by path delimiter char '/' into it's fields.
+        return_name_fields = return_name.split('/')
+        if len(return_name_fields) > 1:
+            # Original device_name has path delimiters in: assume it has a path.
+            return return_name_fields[-1], is_byid
+    # No remove_path request by parameter flag or no path delimiter chars found
+    # in return_name so leave as is and returning.
+    return return_name, is_byid
+
+
+def get_byid_name_map():
+    """
+    Simple wrapper around 'ls -l /dev/disk/by-id' which returns a current
+    mapping of all attached by-id device names to their sdX counterparts.
+    When multiple by-id names are found for the same sdX device then the longest
+    is preferred, or when equal in length then the first listed is used.
+    Intended as a light weight helper for the Dashboard disk activity widget or
+    other non critical components. For critical components use only:
+    get_dev_byid_name() and get_devname() as they contain sanity checks and
+    validation mechanisms and are intended to have more repeatable behaviour but
+    only work on a single device at a time.
+    A single call to this method can provide all current by-id device
+    names mapped to their sdX counterparts with the latter being the index.
+    :return: dictionary indexed (keyed) by sdX type names with associated by-id
+    type names as the values, or an empty dictionary if a non zero return code
+    was encountered by run_command or no by-id type names were encountered.
+    """
+    byid_name_map = {}
+    out, err, rc = run_command([LS, '-l', '/dev/disk/by-id'],
```

Comment:
just a recommendation -- instead of using run_command here, we can simply use python to generate this map. This entire function reduces to

import os, os.path
dirname = '/dev/disk/by-id'
m = {}
for os.listdir(dirname):
    m[os.path.split(os.path.realpath(os.path.join(dirname, d)))[-1]] = d